M2/Macaulay2/tests/normal/memleak5.m2

```diff
@@ -0,0 +1,12 @@
+-- try with:
+--  ctest -T memcheck -R memleak5
+
+R = ZZ[x,y,z];
+I = ideal(9*x^6*y^2*z^2-8*x^2*y^3*z^5,3*x^8*z^2-x^5*y^2*z^3,x^5*y^5-3*x^2*y^6*z^2);
+mingens gb I;
+I = null;
+R = null;
+collectGarbage();
+collectGarbage();
+collectGarbage();
+collectGarbage();
```

Comment:
It seems that this file won't ever produce an error, so there is no reason to run it normally, and it could be put in a new directory called something like `tests/memory-leak-tests`, where people can actually find it.

Reply:
It does if you run it this way:
```
ctest -T memcheck -R memleak5
```
or if you manually run `M2 --check 2` through Valgrind, both of which are reasonable.
There are also other tests with `memleak` prefix in that directory, so I don't see your objection to this file in particular. Perhaps they can all be moved, but I don't want to change both CMake and autotools to handle another directory in this PR.

Reply:
If you decide to create such a directory, I propose naming it just `tests/memory`, and also collecting your various GC tests in there. Specifically `m2/basictests/testgc.c`, `util/gc_demo.c`, and `util/gc_test.c`.

Reply:
I agree with this.  I think that having a separate test directory for such issues is good, but should not be part of this pull request.